src/material-experimental/mdc-list/list-option.ts

```diff
@@ -185,24 +191,30 @@ export class MatListOption extends MatListItemBase implements OnInit, OnDestroy
     this._hostElement.focus();
   }
 
-  _isReversed(): boolean {
-    return this.checkboxPosition === 'after';
+  /** Whether the checkbox should be shown at the given position. */
+  _shouldShowCheckboxAt(position: 'after'|'before'): boolean {
+    return this._selectionList.multiple && this._getCheckboxPosition() === position;
   }
 
-  /** Whether the list-option has a checkbox. */
-  _hasCheckbox() {
-    return this._selectionList.multiple;
+  /** Whether icons and avatars should be shown at the given position. */
+  _shouldShowIconsAndAvatarsAt(position: 'after'|'before'): boolean {
+    return this._getCheckboxPosition() !== position && this._hasIconOrAvatar();
   }
 
   /** Whether the list-option has icons or avatars. */
   _hasIconOrAvatar() {
-    return this._avatars.length || this._icons.length;
+    return this._avatars.length !== 0 || this._icons.length !== 0;
```

Comment:
Not something that was added in this PR, but I think that this may be prone to "changed after checked" errors since the `_avatars` and `_icons` are based on a content query.

Reply:
Yeah, seems out of scope for this PR. I haven't seen any issues with it yet, but should we look into it or keep as is?